Updated references doc

dianeguignard opened the pull request:
Standardization of the format of the references in the file doc/doxygen/references.bib. In particular, I used the conventions

1. "M. Mo" or "R.B. Kellogg" for the names of the authors;
2. Full name for the journal.

I also took the opportunity to add an URL entry for the contributions for which I found a DOI.

Remark: For some references, the DOI entry is commented. If not commented, then the references are not displayed properly, similarly to entry [34] at

https://www.dealii.org/current/doxygen/deal.II/citelist.html

I could not identify the reason for that (example: from references.bib of version 9.3, removing the month entry fixed the issue in some case (e.g., ref "Coleman1963a"), while in others, doing so added the issue (e.g., ref "Holzapfel1996a"))...

bangerth reviewed (approved):
Fantastic, and many thanks for all of that work!

Let's see what happens. This for sure is already much better than what we had before. We can always tweak further if necessary!

bangerth commented:
/rebuild

bangerth commented:
Also thanks for trying to figure out what is going on with references like [34]. Several of us at some point or other have tried to see what might cause this and have failed...

bangerth commented:
Ah, there is the "usual" issue with first-time contributors :-) Here is the error we get from one of the checks:
```

Run ./contrib/utilities/check_indentation.sh
invalid author 'dianeguignard' without firstname and lastname

hint: for possible solutions, consult the webpage:
      https://github.com/dealii/dealii/wiki/Indentation#commit-authorship
Error: Process completed with exit code 2.
```
Can you follow that link and fix the issue?

jppelteret commented on doc/doxygen/references.bib:
```diff
@@ -904,146 +871,144 @@ @article{Riviere1999
 @book{Pelteret2019a,
-  author    = {Pelteret, J-P. V. and Steinmann, P.},
-  publisher = {De Gruyter Mouton},
+  author    = {J.-P. Pelteret and P. Steinmann},
```
May you please change my initials back to `J-P.`. That's my preferential spelling.

jppelteret replied:
Sorry, my comment might have come across as a bit pointed. I see that the other citations for which I'm an author used `J.-P.`, so I guess I should expand a little: Sometimes the publishers give no option for me (and others like me) to specify our preferred initials, so we get "assigned" our initials. When this happens, I use their given spelling because that's how its recorded. But in the event that the bibtex file that the publishers provide use my full name, I choose my preferred spelling. In each of the three publications with attributions to me (Pelteret2019a, Pelteret2018a and davydov2017hp) fall into the latter category, so perhaps I could use the opportunity to ask you to make them consistent with the `J-P.` spelling. Thanks a lot :-)

dianeguignard replied:
Of course. Should I then use this convention for all the compound nouns?

jppelteret replied:
I must admit that I don't have a good answer for you. I'm in a position of privilege that I get to ask for this change (and I got asked similar in https://github.com/dealii/publication-list/issues/280). In general, I would say that we should stick with what's recorded by the publishers. But that's a large task, and should really be something that's done by whoever added these entries in the first place. So if you feel up to checking this, then I think that its one of those tasks that will be unnoticed by most but could be appreciated by one or two of the authors. Otherwise, I'd say that you should leave things as they are (in the hope that they were correctly added in the first place). You've done a lot of work here, which is already greatly appreciated.

dianeguignard replied:
I can definitely check all the entries containing compound nouns. However, I will wait for feedback regarding the abbreviation of first and middle names (see the comment by Marc Fehling below) before doing a new commit.

marcfehling commented:
Thank you very much for cleaning up the references! It certainly helps having a standardization in the bibliography, and you clearly  have fully achieved that!

However, I don't like the abbreviation of first and middle names. To me, this is tantamount to a loss of information. Maybe the others think differently in this respect, but I won't show resistance if I'm outvoted :-)

dianeguignard commented:
> Ah, there is the "usual" issue with first-time contributors :-) Here is the error we get from one of the checks:
> 
> ```
> 
> Run ./contrib/utilities/check_indentation.sh
> invalid author 'dianeguignard' without firstname and lastname
> 
> hint: for possible solutions, consult the webpage:
>       https://github.com/dealii/dealii/wiki/Indentation#commit-authorship
> Error: Process completed with exit code 2.
> ```
> 
> Can you follow that link and fix the issue?

I modified my user name to "Diane Guignard" (for future commits) and updated the information for my last commit. Let me know if this does not work.

marcfehling commented:
> I modified my user name to "Diane Guignard" (for future commits) and updated the information for my last commit. Let me know if this does not work.

You need to update your information for all of your commits. You need to work with `git rebase` or `git filter-branch` in this case as described [here](https://www.git-tower.com/learn/git/faq/change-author-name-email/).

dianeguignard commented:
> Thank you very much for cleaning up the references! It certainly helps having a standardization in the bibliography, and you clearly have fully achieved that!
> 
> However, I don't like the abbreviation of first and middle names. To me, this is tantamount to a loss of information. Maybe the others think differently in this respect, but I won't show resistance if I'm outvoted :-)

The choice of using abbreviation for the first and middle names of the authors was completely random. Of course, I can change the convention and use the full name (or write the first name in full and then use abbreviation for the middle name(s)). Just keep me posted on the decision and I'll make the necessary changes.

jppelteret commented:
My opinion is that @dianeguignard's choice of substituting full names by initials is the pragmatic one. My experience has been that there are quite a number of publications for which there is no record of the full author names. One might guess a few (with the possibility of getting it wrong), but you'd then still end up with a mixture of full and shortened names. If we could use `biblatex` for generating the reference list, then one could (apparently) have it [abbreviate all names for us](https://stackoverflow.com/questions/6612569/automatically-abbreviate-authors-first-names-in-latex#comment93852750_6612569), thereby keeping a true record of all data while still outputting the reference list in a consistent format.

marcfehling commented:
+1 for using `biblatex`.

I don't know whether `doxygen` supports it, but we should address this in a separate PR anyways.

Any other opinions on this?

bangerth commented:
I'm the one who suggested the uniform abbreviation of first names, so am the one to blame. I do think, like @jppelteret , that that's the way to go.

dianeguignard commented:
Should I modify anything (namely add the full names) or should I wait to see if `biblatex` can be used to generate the list of references?

Also I just noticed that I did update the user name of my commits but did not push it (which is now done, hopefully correctly).

marcfehling commented:
> Should I modify anything (namely add the full names) or should I wait to see if biblatex can be used to generate the list of references?

I think we can recover the first names from this PR in the future. So I would leave it as is. We can open an issue to remind us that we _can_ update the first names (but we don't have to).

> Also I just noticed that I did update the user name of my commits but did not push it (which is now done, hopefully correctly).

You merged, which did not overwrite the old commits. You need to `rebase` to clean up your commit history, so that only your two original commits are in it after you branched from the master branch.

tamiko commented:
I'll help @dianeguignard in a minute to rewrite history on this branch :-)

tamiko commented:
@dianeguignard I have rebased and pushed the branch to your repository. It looks clean now.

bangerth reviewed (approved):
Excellent, many thanks!

bangerth commented:
Thanks, and welcome to the list of deal.II authors!
